From 845dd0e5806d1a89d43560b9f075747cf2339fb5 Mon Sep 17 00:00:00 2001
From: Loic Poulain <loic.poulain@intel.com>
Date: Mon, 14 Apr 2014 07:43:50 +0200
Subject: [PATCH] netfilter: Fix proc_set_user() call from xt_quota2

xt_quota2 does not build on Kernel 3.14.
proc_set_user now takes kuid_t & kgid_t as parameters.

Change-Id: Id4d19a94c33910332f4f61da320bd99031a1b7cc
Signed-off-by: Loic Poulain <loic.poulain@intel.com>
---
 net/netfilter/xt_quota2.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- linux.orig/net/netfilter/xt_quota2.c
+++ linux/net/netfilter/xt_quota2.c
@@ -55,7 +55,9 @@ static unsigned int quota_list_perms = S
 static kuid_t quota_list_uid = KUIDT_INIT(0);
 static kgid_t quota_list_gid = KGIDT_INIT(0);
 module_param_named(perms, quota_list_perms, uint, S_IRUGO | S_IWUSR);
-
+module_param_named(uid, quota_list_uid.val, uint, S_IRUGO | S_IWUSR);
+module_param_named(gid, quota_list_gid.val, uint, S_IRUGO | S_IWUSR);
+ 
 #ifdef CONFIG_NETFILTER_XT_MATCH_QUOTA2_LOG
 static void quota2_log(unsigned int hooknum,
 		       const struct sk_buff *skb,
